concent_api/core/payments/bankster.py

```diff
@@ -373,20 +371,11 @@ def settle_overdue_acceptances(
 
             # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
             # have payment_ts < T1
-            t1_is_bigger_than_payments_ts = any(
+            if any(
                 youngest_transaction_timestamp > subtask_results_accepted.payment_ts
                 for subtask_results_accepted in acceptances
-            )
-
-        # Any of the items from list of overdue acceptances
-        # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
-        acceptance_time_overdue = any(
-            cut_off_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
-            for subtask_results_accepted in acceptances
-        )
-
-        if t1_is_bigger_than_payments_ts or acceptance_time_overdue:
-            raise BanksterTimestampError
+            ):
+                raise BanksterTimestampError
```

Comment:
@dybi asked if this part is correct and after looking more closely, it seems to make no sense to me either.

T0 is the timestamp of the oldest acceptance.
T1 is the timestamp of the youngest payment.

In the implementation above Concent will reject provider's case when `t1_is_bigger_than_payments_ts` is `True`, i.e. when there's any acceptance that's older than the youngest payment. So Concent will refuse any case where T0 <= T1. For example a case like this will be rejected outright:

```
     -100 GNT    1 GNT         -100 GNT
       |         |               |
------SRA-------PAY-------------SRA------> [time]
       |         |
       T0        T1
```
even though the right decision is for Concent to pay 199 GNT to the provider.

It might make sense if `PAY` in the above was a forced payment made by Concent (NOTE: I mean forced payment issued in use case 5, not a forced **subtask** payment from use case 3 or 4) because such a payment automatically settles all tasks completed before it (even if it does not cover the whole cost). That's not the case with normal payments.

Reply:
Well, it was like this since first implementation. This is interesting that nobody noticed this before. Please create a separate issue then.